### PR TITLE
Add entropy generation

### LINKED ISSUE #4333: 5.95.5: WiFi Set to "Inactive" on Reboot
## Setup Details
OpenELEC Version: 5.95.5
Build: Generic x86_64
Hardware: Intel NUC DC3217BY w/ Intel 7260 WiFI

## What Did I Do?
A working config on 5.0.8, I use WiFi to connect to MySQL database for library. Use *Wait for Network* to delay Kodi startup while WiFi connects. Upgraded from 5.0.8 to 5.95.5 for testing.

## What Happened?
* After update, no network is active
* Going into OpenELEC Configuration and setting WiFi to *active* turns on the WiFi
* Performing ```systemctl restart kodi``` reconnects to database
* Upon reboot, WiFi is set back to *inactive*
* Not usable in this state. Reverted to 5.0.8 and problem is gone

## What Should Happen?
* WiFi should remain *active* across reboots

## Logs
* Kodi.log: http://sprunge.us/NaGO
* System.log: http://sprunge.us/MHXg
* Hardware.log: http://sprunge.us/PiVV
* Audio.log: http://sprunge.us/EHiN
* Network.log: http://sprunge.us/Adeh
* Input.log: http://sprunge.us/jdAK
* Filesystem.log: http://sprunge.us/FKFM
* Journal_cur.log: http://sprunge.us/gadh
* Journal_prev.log: http://sprunge.us/TXij

For now I've reverted to 5.0.8, but I'd really like to be up to date if possible. If I can provide any additional information to help debug, I'd be happy to.

### PR DESCRIPTION
**Pull request by lrusak**
This hopefully fixes #4333 

Waiting to hear back from @MattDevo 

If this works it should go in for openelec-6.0 aswell

### CODE REVIEW

**Comment by MattDevo** [comment]
checking now against OE master, had to make a slight modification to compile with OE-6.0 (after which it worked) so I'll push and PR those shortly

**Comment by MattDevo** [comment]
@lrusak I needed to remove the [install] section entirely from add-random-at-shutdown.service in order for it to compile for both OE-6.0 and master.  Without that section removed, the build process kept creating /build.xyz/image/usr/lib/systemd/system/poweroff.target as a directory, and then trying to overwrite it with a file of the same name which caused the build process to fail

File: packages/sysutils/entropy/system.d/add-random-at-shutdown.service [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+DefaultDependencies=no
+Before=systemd-poweroff.service systemd-reboot.service systemd-halt.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/lib/entropy/add-random-at-shutdown
+RemainAfterExit=yes
+
+[Install]
+WantedBy=poweroff.target reboot.target halt.target
```
Comment (stefansaraev) [inline packages/sysutils/entropy/system.d/add-random-at-shutdown.service]: our enable_service supports only one WantedBy.

**Comment by MattDevo** [comment]
@stefansaraev all good when limiting to one WantedBy - question is which is best ?

**Comment by stefansaraev** [comment]
none. https://github.com/OpenELEC/OpenELEC.tv/blob/master/config/functions#L227 must be fixed.

```
enable_service () {
  local unit="$1"
  local unit_dir="/usr/lib/systemd/system"
  local target="$2"
  local target_dir=$INSTALL

  [ -f "$target_dir/$unit_dir/$unit" ] || exit 1
  if [ -z "$target" ] ; then
    for target in `grep '^WantedBy' $target_dir/$unit_dir/$unit | cut -f2 -d=` ; do
      if [ -n "$target" ]; then
        mkdir -p ${target_dir}/$unit_dir/${target}.wants
        ln -sf ../${unit} ${target_dir}/$unit_dir/${target}.wants/
      fi
    done
  fi
}
```

not tested but should be enough to support both WantedBy=svc1 svc2 and multi WantedBy lines.

**Comment by MattDevo** [comment]
@stefansaraev yup, that works with all 3 targets

**Comment by stefansaraev** [comment]
this entropy thing smells btw, I am not a crypto expert but what about CONFIG_CRYPTO_JITTERENTROPY=y in kernel config ?

**Comment by MattDevo** [comment]
@stefansaraev believe we'd need to include/make use of rng_tools in order to exploit hardware rng capabilities

**Comment by chewitt** [comment]
Enabling and using HW RNG features in the kernel improves the quality of randomness but doesn't solve the lack of system activity and input to RNG generation that results in a low rate of entropy accumulation and a long delay before /dev/random unlocks. This is a perfectly valid approach to low-activity.

**Comment by MattDevo** [comment]
added the fix foe enable_service: https://github.com/OpenELEC/OpenELEC.tv/pull/4730

**Pull request closed without merging**